### PR TITLE
Fix multiple windows bug and simplify DocumentActivity document handling

### PR DESCRIPTION
**Pull request by harshad1**
This addresses #1570 

Also simplifying document / lifecycle

### CODE REVIEW

File: app/src/main/java/net/gsantner/markor/activity/DocumentActivity.java [hunk 1]
```diff
@@ -85,7 +82,7 @@ public static void launch(Activity activity, File path, Boolean isFolder, Boolea
             intent.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP);
         }
         nextLaunchTransparentBg = (activity instanceof MainActivity);
-        new ActivityUtils(activity).animateToActivity(intent, false, 0).freeContextRef();
+        new ActivityUtils(activity).animateToActivity(intent, false, null).freeContextRef();
```
Comment (gsantner) [inline app/src/main/java/net/gsantner/markor/activity/DocumentActivity.java:85]: this one alone is the window fix right? (0=finish activity)

**Comment by harshad1** [comment]
Yup. Should be null

Jan. 24, 2022 08:50:46 Gregor Santner ***@***.***>:

File: app/src/main/java/net/gsantner/markor/ui/hleditor/HighlightingEditor.java [hunk 2]
```diff
@@ -303,34 +301,18 @@ protected void onSelectionChanged(int selStart, int selEnd) {
...
-
-    public void smoothMoveCursorToLine(final int lineNumber, int... arg0Delay__arg1Duration) {
-        final int delay = Math.max(1, arg0Delay__arg1Duration != null && arg0Delay__arg1Duration.length > 0 ? arg0Delay__arg1Duration[0] : 500);
-        final int duration = Math.max(1, arg0Delay__arg1Duration != null && arg0Delay__arg1Duration.length > 1 ? arg0Delay__arg1Duration[1] : 400);
-
-        this.postDelayed(() -> {
-            String text = getText().toString();
-            int index = StringUtils.getIndexFromLineOffset(text, lineNumber, 0);
-            if (index < 0) {
-                return;
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/ui/hleditor/HighlightingEditor.java:329]: Simplified this logic

File: app/src/main/java/net/gsantner/markor/util/AppSettings.java [hunk 3]
```diff
@@ -354,12 +354,18 @@ public void toggleFavouriteFile(File file) {
...
         }
-        if (!file.equals(getTodoFile()) && !file.equals(getQuickNoteFile())) {
-            setInt(PREF_PREFIX_EDIT_POS_CHAR + file.getAbsolutePath(), pos, _prefCache);
+    }
+
+    public int getLastEditPosition(final String path) {
+        final int _default = 0;
+        if (!fexists(path)) {
+            return _default;
+        } else {
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/util/AppSettings.java:367]: Made this similar to all the other functions

File: app/src/main/java/net/gsantner/markor/activity/DocumentActivity.java (hunk 1 above)
Comment (harshad1) [reply app/src/main/java/net/gsantner/markor/activity/DocumentActivity.java:85]: Yes

**Comment by harshad1** [comment]
This PR has gotten unwieldy and contains a mix of fixes and improvements. I am going to break it up into ~4 smaller PRs which can hopefully be reviewed more easily.

**Comment by gsantner** [comment]
@harshad1 

you can keep it open. doesn't really matter for me, I need time to test and review either way. and if you say OK this single one doesn't break too much, it is quicker then now waiting for you to split up, and waiting for the OK for each once again :D

**Comment by gsantner** [comment]
you can count out the multiple window fix oneliner, I pushed that one directly to master

**Pull request closed without merging**